lib/rucio/db/sqla/migrate_repo/versions/fe1a65b176c9_set_third_party_copy_read_and_write_.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# Copyright 2022 CERN
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors:
+# - Radu Carpa <radu.carpa@cern.ch>, 2022
+
+''' set third_party_copy_read and write fields '''
+
+from alembic import context
+from alembic.op import execute, alter_column  # pylint: disable=no-member
+
+# Alembic revision identifiers
+revision = 'fe1a65b176c9'
+down_revision = '9a45bc4ea66d'
+
+
+def upgrade():
+    '''
+    Upgrade the database to this revision
+    '''
+    if context.get_context().dialect.name in ['oracle', 'mysql', 'postgresql']:
+        schema = context.get_context().version_table_schema if context.get_context().version_table_schema else ''
+        schema_prefix = schema + '.' if schema else ''
+        execute('UPDATE ' + schema_prefix + 'rse_protocols SET third_party_copy_read=third_party_copy WHERE third_party_copy_read is NULL')
```

Comment:
Do we technically need this? Since the functionality above is backwards compatible? Thus for the people who still just sync in `third_party_copy` it would still be working?

Reply:
As implemented right now, third_party_copy isn't used anymore anyway. everything relies on _read/_write counterparts. When somebody sets third_party_copy using the cli/api, its value is copied into _read/_write, which are then used. So, yes, in the current form of the pull request, this is necessary. But I'm open to suggestions to make it differently

Reply:
I think the way you currently have it is probably better. I am wondering, if  we should just remove the `third_party_copy` column then entirely. People can still sync in a `third_party_copy` value, which will just set both `read/write` or the `read` and `write` value individually.

Reply:
I propose to do it at next major release. Just to ensure in the meantime that none of corner cases was forgotten.

Reply:
Having a possibility for easy roll-back seems important to me. Removing the column breaks such possibility